circe-yaml-v12/src/main/scala/io/circe/yaml/v12/ParserImpl.scala

```diff
@@ -0,0 +1,137 @@
+   * @param yaml
+   * @return
+   */
+  def parse(yaml: Reader): Either[ParsingFailure, Json] = for {
+    parsed <- parseSingle(yaml)
+    json <- yamlToJson(parsed)
+  } yield json
+
+  def parse(yaml: String): Either[ParsingFailure, Json] =
+    parse(new StringReader(yaml))
+
+  def parseDocuments(yaml: Reader): Stream[Either[ParsingFailure, Json]] = parseStream(yaml).map(yamlToJson)
+  def parseDocuments(yaml: String): Stream[Either[ParsingFailure, Json]] = parseDocuments(new StringReader(yaml))
+
+  private[this] def asScala[T](ot: Optional[T]): Option[T] =
+    if (ot.isPresent) Some(ot.get()) else None
+
+  private[this] def createComposer(reader: Reader) =
+    new Composer(settings, new org.snakeyaml.engine.v2.parser.ParserImpl(settings, new StreamReader(settings, reader)))
+
+  private[this] def parseSingle(reader: Reader): Either[ParsingFailure, Node] =
+    Either.catchNonFatal {
+      val composer = createComposer(reader)
+      asScala(composer.getSingleNode)
+    } match {
+      case Left(err)          => Left(ParsingFailure(err.getMessage, err))
+      case Right(None)        => Left(ParsingFailure("no document found", new RuntimeException("no document found")))
+      case Right(Some(value)) => Right(value)
+    }
+
+  private[this] def parseStream(reader: Reader) =
+    createComposer(reader).asScala.toStream
+
+  private[this] object CustomTag {
+    def unapply(tag: Tag): Option[String] = if (!tag.getValue.startsWith(Tag.PREFIX))
+      Some(tag.getValue)
+    else
+      None
+  }
+
+  private[this] class FlatteningConstructor(settings: LoadSettings) extends StandardConstructor(settings) {
+    def flatten(node: MappingNode): MappingNode = {
+      flattenMapping(node)
+      node
+    }
+
+    def construct(node: ScalarNode): Object = super.construct(node) // to make the method public
+  }
+
+  private[this] def yamlToJson(node: Node): Either[ParsingFailure, Json] = {
+    // Isn't thread-safe internally, may hence not be shared
+    val flattener: FlatteningConstructor = new FlatteningConstructor(settings)
+
+    def convertScalarNode(node: ScalarNode) = Either
+      .catchNonFatal(node.getTag match {
+        case Tag.INT if node.getValue.startsWith("0x") || node.getValue.contains("_") =>
+          Json.fromJsonNumber(flattener.construct(node) match {
+            case int: Integer         => JsonLong(int.toLong)
+            case long: java.lang.Long => JsonLong(long)
+            case bigint: java.math.BigInteger =>
+              JsonDecimal(bigint.toString)
+            case other => throw new NumberFormatException(s"Unexpected number type: ${other.getClass}")
+          })
+        case Tag.INT | Tag.FLOAT =>
+          JsonNumber.fromString(node.getValue).map(Json.fromJsonNumber).getOrElse {
+            throw new NumberFormatException(s"Invalid numeric string ${node.getValue}")
+          }
+        case Tag.BOOL =>
+          Json.fromBoolean(flattener.construct(node) match {
```

Comment:
See above on how snakeyaml-engine "only supports JSON schema". We have no need for calling`construct` method on literals, as snakeyaml-engine doesn't support any more literal formats than JSON itself already does.